exercises/matthew_cumber/ex01_basics/src/main.cpp

```diff
@@ -6,29 +8,31 @@
 #include <sstream>
 #include <vector>
 #include <ctype.h>
-#include <regex>
-#include "UniqueWord.h"
-
-std::string PATH = ""; // Will store the path of the file to be read
 
-/** Tests if a word has already been found the increments it's count
+/** Checks if a word already exists in the vector or words, then increments it's count if found,
+ * otherwise create a new word and add to vector
+ *
  * @param uniqueWords The vector of existing unique words found
- * @param word The word that is being tested
- * @return A boolean, true if a word already exists, false otherwise
+ * @param word The word being tested
+ * @return No return value 
  */
-bool inUniqueWords(std::vector<UniqueWord> &uniqueWords, const std::string &word)
+void incrementCountOrAddNewWord(std::vector<UniqueWord> &uniqueWords, const std::string &word)
 {
-  // Loop unique words, if word exists increment it's count
-  for(auto &w : uniqueWords)
+  // Attempt to find word in the vector uniqueWords
+  auto it = std::find_if(uniqueWords.begin(), uniqueWords.end(), [&word](const UniqueWord &obj){return obj.getWord() == word;});
```

Comment:
Thanks for changing. Looking at this again my argument for changing on my previous review wasn't quite right.
If you had been using a "raw" for loop that accessed the items in the vector using an index eg uniqueWords[i] then not all of the STL containers support random access like this. However you weren't...
So the benefit of this change is actually about performance - the STL find function on some platforms may well be quicker than writing your own loop